tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/Sources.cs

```diff
@@ -35,6 +35,48 @@ void BuildInstrumentationDefinitions(StringBuilder sb, List<CallTargetDefinition
             }};");
             }
 
+            static void BuildInstrumentedAssemblies(StringBuilder sb, IReadOnlyCollection<CallTargetDefinitionSource> orderedDefinitions)
+            {
+                var hashSet = new HashSet<string>();
+                foreach (var orderedDefinition in orderedDefinitions)
+                {
+                    if (!IsKnownAssemblyPrefix(orderedDefinition.AssemblyName))
+                    {
+                        hashSet.Add(orderedDefinition.AssemblyName);
+                    }
+                }
+
+                sb.Append(
+                    $$"""
+                    #if NETCOREAPP
+                                    new HashSet<string>({{hashSet.Count}}, StringComparer.Ordinal)
+                    #else
+                                    new HashSet<string>(StringComparer.Ordinal)
+                    #endif
+                                    { 
+                    """);
+
+                foreach (var assembly in hashSet.OrderBy(x => x))
+                {
+                    sb.Append('"')
+                      .Append(assembly)
+                      .Append("\", ");
+                }
+
+                sb.AppendLine("};");
+                return;
+
+                // NOTE: Keep this in sync with ExceptionRedactor.IsKnownAssemblyPrefix()
+                static bool IsKnownAssemblyPrefix(string assemblyName)
+                {
+                    return assemblyName.StartsWith("Datadog.", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("mscorlib,", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("Microsoft.", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("System.", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("Azure.", StringComparison.Ordinal);
```

Comment:
A bit late as I could have said it yesterday, but we could have AWSSDK here as we're adding Azure. Anyway, it's handled but felt more consistent. 
This is really nitpicky, feel free to ignore.

Reply:
tbh, I'm a bit torn on the Azure one... it feels a bit arbitrary. But yeah we could add that. But then there's lots more we _could_ add too 😅 We could just maintain a manual list of prefixes instead 🤷‍♂️

Reply:
YEs I don't have stong opinions. I assume that the part of the stack we would miss in case there are issues won't be a game changer anyway. So either way is good. I was just mentionning as I thought it was unconsistent. But I'm fine as is and we can revisit if we find stronger arguments later